File: apps/web/src/features/sidebar/relativeTime.ts
// Compact relative time like "9h", "2d", "now" — port of the Swift
// RelativeTime.short helper.
export function relativeTimeShort(iso: string, now: Date = new Date()): string {
  const date = new Date(iso)
  const seconds = Math.max(0, (now.getTime() - date.getTime()) / 1000)
  if (seconds < 60) return "now"
  if (seconds < 3600) return `${Math.floor(seconds / 60)}m`
  if (seconds < 86_400) return `${Math.floor(seconds / 3600)}h`
  return `${Math.floor(seconds / 86_400)}d`
}

File: apps/macos/Codevisor/Features/Sidebar/RelativeTime.swift
import Foundation

/// Compact relative time like "9h", "2d", "now".
enum RelativeTime {
  static func short(from date: Date, now: Date = Date()) -> String {
    let seconds = max(0, now.timeIntervalSince(date))
    switch seconds {
    case ..<60: return "now"
    case ..<3600: return "\(Int(seconds / 60))m"
    case ..<86_400: return "\(Int(seconds / 3600))h"
    default: return "\(Int(seconds / 86_400))d"
    }
  }
}
